src/ServantElm.hs

```diff
@@ -166,6 +238,118 @@ mkRequest request =
         Nothing -> error "mkHttpRequest: no reqReturnType?"
+
+    body =
+      case request ^. reqBody of
+        Just elmTypeExpr ->
+          "Http.jsonBody" <+> parens ((typeRefEncoder . definitionToRef) elmTypeExpr <+> bodyValue)
+        Nothing ->
+          "Http.emptyBody"
+
+typeRefDecoder :: TypeRef -> Doc ann
+typeRefDecoder (RefCustom TypeName {unTypeName}) = "decode" <> pretty (T.takeWhile (/= ' ') unTypeName)
+typeRefDecoder (RefPrim elmPrim) = case elmPrim of
+  ElmUnit -> parens "JD.map (always ()) (JD.list JD.string)"
+  ElmNever -> parens "JD.fail \"Never is not possible\""
+  ElmBool -> "JD.bool"
+  ElmChar -> "elmStreetDecodeChar"
+  ElmInt -> "JD.int"
+  ElmFloat -> "JD.float"
+  ElmString -> "JD.string"
+  ElmTime -> "Iso.decoder"
+  ElmMaybe t ->
+    parens $
+      "JD.maybe"
+        <+> typeRefDecoder t
+  ElmResult l r ->
+    parens $
+      "elmStreetDecodeEither"
+        <+> typeRefDecoder l
+        <+> typeRefDecoder r
+  ElmPair a b ->
+    parens $
+      "elmStreetDecodePair"
+        <+> typeRefDecoder a
+        <+> typeRefDecoder b
+  ElmTriple a b c ->
+    parens $
+      "elmStreetDecodeTriple"
+        <+> typeRefDecoder a
+        <+> typeRefDecoder b
+        <+> typeRefDecoder c
+  ElmList l -> parens $ "JD.list" <+> typeRefDecoder l
+
+typeRefEncoder :: TypeRef -> Doc ann
+typeRefEncoder (RefCustom TypeName {unTypeName}) = "encode" <> pretty (T.takeWhile (/= ' ') unTypeName)
+typeRefEncoder (RefPrim elmPrim) = case elmPrim of
+  ElmUnit -> "always <| JE.list identity []"
+  ElmNever -> "never"
+  ElmBool -> "JE.bool"
+  ElmChar -> "JE.string << String.fromChar"
+  ElmInt -> "JE.int"
+  ElmFloat -> "JE.float"
+  ElmString -> "JE.string"
+  ElmTime -> "Iso.encode"
+  ElmMaybe t ->
+    "elmStreetEncodeMaybe"
+      <+> parens (typeRefEncoder t)
+  ElmResult l r ->
+    "elmStreetEncodeEither"
+      <+> parens (typeRefEncoder l)
+      <+> parens (typeRefEncoder r)
+  ElmPair a b ->
+    "elmStreetEncodePair"
+      <+> parens (typeRefEncoder a)
+      <+> parens (typeRefEncoder b)
+  ElmTriple a b c ->
+    "elmStreetEncodeTriple"
+      <+> parens (typeRefEncoder a)
+      <+> parens (typeRefEncoder b)
+      <+> parens (typeRefEncoder c)
+  ElmList l -> "JE.list" <+> parens (typeRefEncoder l)
+
+generateElmModule ::
+  ( HasForeign LangElm ElmDefinition api,
+    GenerateList ElmDefinition (Foreign ElmDefinition api)
+  ) =>
+  Settings ->
+  Proxy api ->
+  IO ()
+generateElmModule Settings {settingsDirectory, settingsModule} api =
+  TIO.writeFile filePath (showDoc moduleFile)
+  where
+    moduleName :: Doc ann
+    moduleName = "Core.Generated.ElmQueries"
+
+    moduleHeader :: Doc ann
+    moduleHeader = "module" <+> moduleName <+> "exposing" <+> parens ".."
+
+    queries :: Doc ann
+    queries = vsep (elmForAPI api)
+
+    imports :: Doc ann
+    imports =
+      vsep $
+        map
+          ("import" <+>)
+          [ "Core.Generated.Decoder" <+> "exposing" <+> parens "..",
```

Comment:
This is taken from `elm-street`, right? You should add a comment, so that we know what we should ask `elm-street` to expose from their api eventually.

Reply:
These module names shouldn't be hardcoded, they should come from the settings.